CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt

```diff
@@ -301,6 +298,10 @@ class CameraFragment : Fragment() {
                 // camera provides access to CameraControl & CameraInfo
                 camera = cameraProvider.bindToLifecycle(
                         this, cameraSelector, preview, imageCapture, imageAnalyzer)
+
+                // Attach the viewfinder's surface provider to preview use case
+                previewView.preferredImplementationMode = PreviewView.ImplementationMode.TEXTURE_VIEW
```

Comment:
It is recommended to leave the preferred implementation mode as the default, please remove this line

Reply:
I put this on to respect the doc: 
https://developer.android.com/jetpack/androidx/releases/camera#camera-view-1.0.0-alpha09

Because ImplementationMode.SURFACE_VIEW can cause bugs on some devices :(

Reply:
By leaving it to the default CameraX will choose whatever is most appropriate for the it's running on device. I believe the issues described in those release notes have now been fixed, and in addition to that the implementation mode is only a preference. Even if you choose `ImplementationMode.SURFACE_VIEW` as a preference, CameraX might still use `TextureView` if the device can't properly implement camera preview using `SurfaceView`.